dxp/react_rendering.md

```diff
@@ -0,0 +1,148 @@
+2. `componentId`: the `componentId` that will be given to the component's JavaScript as an initial property. Note that this field is only used if the `data` map passed to `renderReact` does **NOT** contain a `componentId` property.
+3. `dependencies`: this is deprecated; don't use it 🧐.
+4. `positionInline`: a flag to render the JavaScript directly in the current position of the HTML response or defer it to the bottom of the page. By default `false`.
+5. `propsTransformer`: optional name of an AMD module exporting a function to transform the properties before they are passed to the component.
+
+### How the React component is rendered
+
+Say you run the following code:
+
+```java
+ComponentDescriptor componentDescriptor =
+	new ComponentDescriptor(
+		"my-bundle@1.0.0/component/button", "myButton", null, false,
+		"my-bundle@1.0.0/propsTransformer");
+
+Map<String, Object> data = new HashMap<>();
+
+data.put("color", "red");
+
+reactRenderer.renderReact(
+	componentDescriptor, data, httpServletRequest,
+	httpServletResponse.getWriter());
+```
+
+This will cause HTML like this to be rendered in the current position of the `HttpServletResponse`:
+
+```html
+<div id="XJAZ"></div>
+```
+
+(where `XJAZ` is a random placeholder id).
+
+And JavaScript like this to be appended at the end of the page:
+
+```javascript
+Liferay.Loader.require(
+	'portal-template-react-renderer-impl@1.2.3/render.es',
+	'my-bundle@1.0.0/component/button',
+	'my-bundle@1.0.0/propsTransformer'
+	function (render, renderFunction, propsTransformer) {
+		render(renderFunction, propsTranformer({'color': 'red'}));
+	}
+);
+```
+
+If you are curious about what `portal-template-react-renderer-impl@1.2.3/render.es` contains, you can see its content in [liferay-portal's source code](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-template/portal-template-react-renderer-impl/src/main/resources/META-INF/resources/render.es.js). It is simply a helper to connect the generated JavaScript and the React component but, in general, it's a level of indirection to be able to add any extra infrastructure code we may need in the future.
+
+## react:component Tag
+
+-   OSGi bundle: `com.liferay.frontend.taglib.react`
+-   Taglib URI: `http://liferay.com/tld/react`
+-   Tag: `<react:component>`
+
+This tag is a wrapper for the low-level React renderer service. It leverages OSGi service `com.liferay.portal.template.react.renderer.ReactRenderer` to render the HTML and JavaScript.
+
+The tag has five attributes (all optional but the `module` one):
+
+1. `componentId`: the component identifier (which can be used to reference the React component with `Liferay.component(componentId)`).
```

Comment:
Small nit. Remove the `(...)` and simplify the sentence maybe like: `the component identifier. It can be used to...`

Reducing subordination could make this easier to read.

Reply:
Subordination is what distinguishes real writers from mere amateurs... :unamused:

Reply:
Done.